### ISSUE TITLE
Bounds and packages removed from LTS-11.13

### ISSUE THREAD

**Issue by mihaimaruseac** [commercialhaskell/lts-haskell#132]
This issues serves as documentation for why some packages were removed from LTS-11.13 although they were present before.

See also #131, upgrading `primitive` caused a few packages and the packages depending on them to be dropped:

prim-array (not present) depended on by:
- [ ] quickcheck-classes-0.3.3 (-any). No maintainer. @andrewthad. Used by: library

quickcheck-classes (not present) depended on by:
- [ ] ip-1.1.2 (==0.3.*). No maintainer. @andrewthad. Used by: test-suite

ip (not present) depended on by:
- [ ] attoparsec-ip-0.0.1 (>=1.1.0). Athan Clark <athan.clark@gmail.com> @athanclark. @athanclark. Used by: library
- [ ] attoparsec-uri-0.0.4 (>=1.1.0). Athan Clark <athan.clark@gmail.com> @athanclark. @athanclark. Used by: library

attoparsec-uri (not present) depended on by:
- [ ] markup-4.0.4 (>=0.0.4). Athan Clark <athan.clark@gmail.com> @athanclark. @athanclark. Used by: library
- [ ] urlpath-8.0.1 (>=0.0.4). Athan Clark <athan.clark@gmail.com> @athanclark. @athanclark. Used by: library

`prim-array` has the following errors when building:

```
...
    |
275 | indexPrimArray (PrimArray arr#) (I# i#) = indexByteArray# arr# i#
    |                 ^^^^^^^^^

src/Data/Primitive/PrimArray.hs:277:40: error:
    Ambiguous occurrence ‘PrimArray’
    It could refer to either ‘Data.Primitive.PrimArray’,
                             imported from ‘Data.Primitive’ at src/Data/Primitive/PrimArray.hs:42:1-21
                             (and originally defined in ‘Data.Primitive.PrimArray’)
                          or ‘Data.Primitive.PrimArray.PrimArray’,
                             defined at src/Data/Primitive/PrimArray.hs:51:1
    |
277 | sizeofPrimArray :: forall a. Prim a => PrimArray a -> Int
    |                                        ^^^^^^^^^

src/Data/Primitive/PrimArray.hs:279:18: error:
    Ambiguous occurrence ‘PrimArray’
    It could refer to either ‘Data.Primitive.PrimArray’,
                             imported from ‘Data.Primitive’ at src/Data/Primitive/PrimArray.hs:42:1-21
                             (and originally defined in ‘Data.Primitive.PrimArray’)
                          or ‘Data.Primitive.PrimArray.PrimArray’,
                             defined at src/Data/Primitive/PrimArray.hs:51:20
    |
279 | sizeofPrimArray (PrimArray arr#) = I# (quotInt# (sizeofByteArray# arr#) (sizeOf# (undefined :: a)))
    |
```

`unordered-intmap` is another package which failed to build with the upgrade to `primitive`:

```
Building library for unordered-intmap-0.1.0.0..
[1 of 3] Compiling Data.Primitive.SmallArray.Extra ( src/Data/Primitive/SmallArray/Extra.hs, dist/build/Data/Primitive/SmallArray/Extra.o )

src/Data/Primitive/SmallArray/Extra.hs:32:8-20: error:
    Ambiguous occurrence ‘runSmallArray’
    It could refer to either ‘Data.Primitive.SmallArray.runSmallArray’,
                             imported from ‘Data.Primitive.SmallArray’ at src/Data/Primitive/SmallArray/Extra.hs:23:1-32
                          or ‘Data.Primitive.SmallArray.Extra.runSmallArray’,
                             defined at src/Data/Primitive/SmallArray/Extra.hs:44:1
   |
32 |     in runSmallArray $ do
   |        ^^^^^^^^^^^^^
```

Finally, `bytestring-strict-builder` was not updated to the latest version due to build failure

```
library/ByteString/StrictBuilder/Population.hs:24:3: error:
    ‘<>’ is not a (visible) method of class ‘Semigroup’
   |
24 |   (<>) = mappend
   |   ^^^^
```

Full list of extra constraints:

* `--add-package 'web3'` (#87)
* `--expect-test-failure web3` (#87)
* `--constraint 'primitive > 0.6.3'` (#131)
* `--constraint 'bytestring-strict-builder < 0.4.5.1'` (see above)
* `--constraint 'prim-array < 0'` (see above, `primitive`)
* `--constraint 'unordered-intmap < 0'` (see above, `primitive`)
* `--constraint 'quickcheck-classes < 0'` (from `prim-array`)
* ` --constraint 'ip < 0'` (from `quickheck-classes`)
* `--constraint 'attoparsec-ip < 0'` (from `ip`)
* `--constraint 'attoparsec-uri < 0'` (from `ip`)
* `--constraint 'markup < 0'` (from `attoparsec-uri`)
* `--constraint 'urlpath < 0'` (from `attoparsec-uri`)

**Comment by mihaimaruseac** [comment]
Just realized that `ip` could have been added with skipped tests, we should do that on the next LTS and include the packages depending on it which were dropped:

```
* ` --constraint 'ip < 0'` (from `quickheck-classes`)
* `--constraint 'attoparsec-ip < 0'` (from `ip`)
* `--constraint 'attoparsec-uri < 0'` (from `ip`)
* `--constraint 'markup < 0'` (from `attoparsec-uri`)
* `--constraint 'urlpath < 0'` (from `attoparsec-uri`)
```

**Comment by bergmark** [comment]
Looks like the remainder was fixed in LTS 11.14
